plugins/modules/ndo_template.py

```diff
@@ -247,12 +244,13 @@ def main():
                 set_service_device_template_payload(payload, template_type, tenant_id, site_ids)
 
             if not module.check_mode:
-                response = mso.request(mso_template.templates_path, method="POST", data=payload)
-                payload["templateId"] = response.get("templateId")
+                mso.existing = mso.request(mso_template.templates_path, method="POST", data=payload)
+                payload["templateId"] = mso.existing.get("templateId")
```

Comment:
can you check the response for tenant and its corresponding id in the response also?